File: src/main/java/com/wanted/preonboarding/ticket/application/policy/discount/DefaultDiscountPolicy.java
package com.wanted.preonboarding.ticket.application.policy.discount;

public class DefaultDiscountPolicy implements DiscountPolicy {

	@Override
	public Long calculateAmount(int performanceAmount, Long reservationAmount) {
		long result = reservationAmount - performanceAmount;
		if (result < 0) {
			throw new IllegalArgumentException("잔액이 부족합니다.");
		}
		return result;
	}
}

File: src/main/kotlin/com/wanted/preonboarding/ticket/application/discount/DefaultDiscountPolicy.kt
package com.wanted.preonboarding.ticket.application.discount

import com.wanted.preonboarding.ticket.domain.Performance
import com.wanted.preonboarding.ticket.domain.SeatInfo
import com.wanted.preonboarding.ticket.domain.UserInfo
import org.springframework.stereotype.Component

@Component
class DefaultDiscountPolicy : DiscountPolicy {
    override fun isAcceptable(
        userInfo: UserInfo,
        seatInfo: SeatInfo,
        performance: Performance,
    ): Boolean {
        return true
    }

    override fun discountRate(): Double {
        return 0.01
    }
}
